app/components/fade.js

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {Animated} from 'react-native';
+
+export const FADE_DURATION = 300;
+
+export default class Fade extends Component {
+    static propTypes = {
+        children: PropTypes.node.isRequired,
+        style: PropTypes.object,
+        visible: PropTypes.bool.isRequired,
+        disabled: PropTypes.bool.isRequired,
```

Comment:
What does this component use the `disabled` prop for? If it's used so that the child re-renders when it's enabled or disabled, I'd prefer to just check `props.children` in `shouldComponentUpdate` instead.

Reply:
I didn't realize I was duplicating logic from the send button! Took it out. It looks so much cleaner. Thank you for the great review!